integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-form/dp-payment-form.component.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { DpPaymentRequest } from './../../../models/dp-checkout.model';
+import {
+  MockTranslatePipe,
+  WindowRef,
+  GlobalMessageService,
+  GlobalMessageType,
+} from '@spartacus/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { DpPaymentFormComponent } from './dp-payment-form.component';
+import { DpCheckoutPaymentService } from '../../../facade';
+import { Observable, of } from 'rxjs';
+import { Component } from '@angular/core';
+import { StoreModule } from '@ngrx/store';
+
+const postUrl = 'https://dummy.url';
+const mockDpPaymentRequest: DpPaymentRequest = {
+  url: postUrl,
+  signature: 'ASDFGHJKE456789',
+  sessionId: 'QWERTYUIOPASDFGHJKLZXCVBNM',
+};
+
+const mockWinRef: WindowRef = {
+  nativeWindow: {
+    location: { href: 'previous-url' },
+  },
+} as WindowRef;
+
+class MockDpCheckoutPaymentService
+  implements Partial<DpCheckoutPaymentService> {
+  getCardRegistrationDetails(): Observable<DpPaymentRequest | undefined> {
+    return of({});
+  }
+}
+
+@Component({
+  selector: 'cx-spinner',
+  template: '',
+})
+class MockSpinnerComponent {}
+
+describe('DpPaymentFormComponent', () => {
+  let component: DpPaymentFormComponent;
+  let fixture: ComponentFixture<DpPaymentFormComponent>;
+  let dpPaymentService: MockDpCheckoutPaymentService;
+  let winRef: WindowRef;
+  let msgService: GlobalMessageService;
+
+  beforeEach(
+    async () => {
+    await
+     TestBed.configureTestingModule({
+      declarations: [
+        DpPaymentFormComponent,
+        MockTranslatePipe,
+        MockSpinnerComponent,
+      ],
+      imports: [StoreModule.forRoot({}),],
```

Comment:
I think StoreModule might not be needed here

Reply:
removing it gives this error:
NullInjectorError: R3InjectorError(DynamicTestModule)[GlobalMessageService -> Store -> Store]: 
  NullInjectorError: No provider for Store!